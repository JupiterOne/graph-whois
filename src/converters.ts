import { Domain } from "./DomainProviderClient";

import { DOMAIN_ENTITY_CLASS, DOMAIN_ENTITY_TYPE, DomainEntity } from "./types";
import getTime from "./util/getTime";

export function createDomainEntity(domain: Domain): DomainEntity {
  return {
    _key: `internet_domain_${domain.name}`,
    _type: DOMAIN_ENTITY_TYPE,
    _class: DOMAIN_ENTITY_CLASS,
    _rawData: [{ name: "default", rawData: domain }],
    id: domain.id,
    name: domain.name,
    displayName: domain.name,
    domainName: domain.name,
    expiresOn: getTime(domain.expiration_date),
    updatedOn: getTime(domain.updated_date),
    createdOn: getTime(domain.creation_date),
    whoisServer: domain.whois_server,
    registrar: domain.registrar,
    status: domain.status,
    contactEmails: domain.emails,
    nameservers: domain.nameservers,
  };
}
