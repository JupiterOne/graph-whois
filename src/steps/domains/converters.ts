import {
  createIntegrationEntity,
  parseTimePropertyValue,
} from '@jupiterone/integration-sdk-core';
import { Entities } from '../../constants';
import { Domain } from '../../DomainProviderClient';

export function createDomainEntity(domain: Domain) {
  return createIntegrationEntity({
    entityData: {
      source: domain,
      assign: {
        _key: `internet_domain_${domain.name}`,
        _type: Entities.DOMAIN._type,
        _class: Entities.DOMAIN._class,
        id: domain.id,
        name: domain.name,
        displayName: domain.name,
        domainName: domain.name,
        expiresOn: parseTimePropertyValue(domain.expiration_date),
        updatedOn: parseTimePropertyValue(domain.updated_date),
        createdOn: parseTimePropertyValue(domain.creation_date),
        whoisServer: domain.whois_server,
        registrar: domain.registrar,
        status: domain.status,
        contactEmails: Array.isArray(domain.emails)
          ? domain.emails
          : [domain.emails],
        nameservers: Array.isArray(domain.nameservers)
          ? domain.nameservers
          : [domain.nameservers],
      },
    },
  });
}
