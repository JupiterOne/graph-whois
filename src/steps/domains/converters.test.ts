import { createDomainEntity } from './converters';

describe('#createDomainEntity', () => {
  test('should convert to entity', () => {
    expect(
      createDomainEntity({
        id: '2138514_DOMAIN_COM-VRSN',
        whois_server: 'whois.markmonitor.com',
        updated_date: '2019-09-09T15:39:04Z',
        creation_date: '1997-09-15T04:00:00Z',
        expiration_date: '2028-09-14T04:00:00Z',
        registrar: 'MarkMonitor Inc.',
        emails: 'abusecomplaints@markmonitor.com',
        status:
          'serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited',
        nameservers: 'NS4.GOOGLE.COM',
        contact: {
          registrant: {},
          technical: {},
          admin: {},
          billing: {},
        },
        name: 'google.com',
      }),
    ).toMatchSnapshot();
  });
});
