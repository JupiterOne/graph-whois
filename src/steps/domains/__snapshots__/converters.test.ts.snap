// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createDomainEntity should convert to entity 1`] = `
{
  "_class": [
    "Domain",
  ],
  "_key": "internet_domain_google.com",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "contact": {
          "admin": {},
          "billing": {},
          "registrant": {},
          "technical": {},
        },
        "creation_date": "1997-09-15T04:00:00Z",
        "emails": "abusecomplaints@markmonitor.com",
        "expiration_date": "2028-09-14T04:00:00Z",
        "id": "2138514_DOMAIN_COM-VRSN",
        "name": "google.com",
        "nameservers": "NS4.GOOGLE.COM",
        "registrar": "MarkMonitor Inc.",
        "status": "serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited",
        "updated_date": "2019-09-09T15:39:04Z",
        "whois_server": "whois.markmonitor.com",
      },
    },
  ],
  "_type": "internet_domain",
  "active": undefined,
  "contactEmails": [
    "abusecomplaints@markmonitor.com",
  ],
  "createdOn": 874296000000,
  "displayName": "google.com",
  "domainName": "google.com",
  "expiresOn": 1852516800000,
  "id": "2138514_DOMAIN_COM-VRSN",
  "name": "google.com",
  "nameservers": [
    "NS4.GOOGLE.COM",
  ],
  "registrar": "MarkMonitor Inc.",
  "status": "serverUpdateProhibited https://icann.org/epp#serverUpdateProhibited",
  "updatedOn": 1568043544000,
  "whoisServer": "whois.markmonitor.com",
}
`;
